pkg/knative/describer.go

```diff
@@ -26,54 +25,60 @@ func NewDescriber(verbose bool) *Describer {
 // escaped. Therefore as a knative (kube) implementation detal proper full
 // names have to be escaped on the way in and unescaped on the way out. ex:
 // www.example-site.com -> www-example--site-com
-func (d *Describer) Describe(ctx context.Context, name, namespace string) (description fn.Instance, err error) {
+func (d *Describer) Describe(ctx context.Context, name, namespace string) (*fn.Instance, error) {
```

Comment:
My initial reaction here is that we should not have the possibility of a nil return, but instead have a typed error such as ErrFunctionNotFound the caller can selectively decide to watch for, returning an empty struct if the given name does not exist in the given namespace.